internal/test/builder/builders.go

```diff
@@ -485,6 +486,12 @@ func (m *MachineDeploymentClassBuilder) WithStrategy(s *clusterv1.MachineDeploym
 	return m
 }
 
+// WithNamingStrategy sets the Strategy for the MachineDeploymentClassBuilder.
```

Comment:
Same for MP (pending merge/rebase) (looks like we don't have anything for ControlPlaneClass)